src/components/SettlementButton/AnimatedSettlementButton.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useEffect, useState} from 'react';
+import Animated, {runOnJS, useAnimatedStyle, useSharedValue, withDelay, withTiming} from 'react-native-reanimated';
+import Text from '@components/Text';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import SettlementButton from '.';
+import {SettlementButtonProps} from './types';
+
+type AnimatedSettlementButtonProps = SettlementButtonProps & {
+    isVisible: boolean;
+};
+
+function AnimatedSettlementButton({isVisible, ...settlementButtonProps}: AnimatedSettlementButtonProps) {
+    const styles = useThemeStyles();
+    const buttonScale = useSharedValue(1);
+    const buttonOpacity = useSharedValue(1);
+    const paymentCompleteTextScale = useSharedValue(0);
+    const paymentCompleteTextOpacity = useSharedValue(1);
+    const height = useSharedValue<number>(variables.componentSizeNormal);
+    const buttonStyles = useAnimatedStyle(() => ({
+        transform: [{scale: buttonScale.value}],
+        opacity: buttonOpacity.value,
+    }));
+    const paymentCompleteTextStyles = useAnimatedStyle(() => ({
+        transform: [{scale: paymentCompleteTextScale.value}],
+        opacity: paymentCompleteTextOpacity.value,
+        position: 'absolute',
+        alignSelf: 'center',
+    }));
+
+    const containerStyles = useAnimatedStyle(() => ({
+        height: height.value,
+        justifyContent: 'center',
+        overflow: 'hidden',
+    }));
+
+    const [isAnimationRunning, setIsAnimationRunning] = useState(false);
+
+    const resetAnimation = () => {
+        buttonScale.value = 1;
+        buttonOpacity.value = 1;
+        paymentCompleteTextScale.value = 0;
+        paymentCompleteTextOpacity.value = 1;
+        height.value = variables.componentSizeNormal;
+    }
+
+    useEffect(() => {
+        if (!isAnimationRunning) {
+            resetAnimation();
+            return;
+        }
```

Comment:
I reset the animation here in case of a pay error.

https://github.com/user-attachments/assets/4b219af9-c4e7-423c-b823-a937c8015797